web-common/src/features/dashboards/time-dimension-details/Pivot.svelte

```diff
@@ -0,0 +1,216 @@
+  export let renderCell: (rowIdx: number, colIdx: number) => any;
+  export let height: number;
+
+  function range(n: number) {
+    return new Array(n).fill(0).map((d, i) => i);
+  }
+
+  let container: HTMLDivElement;
+  $: rowVirtualizer = createVirtualizer({
+    count: rowCt,
+    getScrollElement: () => container,
+    estimateSize: getRowSize,
+    overscan: 10,
+  });
+
+  let virtualRows: VirtualItem[] = [];
+  let totalVerticalSize = 0;
+  let paddingTop = 0;
+  let paddingBottom = 0;
+  $: {
+    virtualRows = $rowVirtualizer?.getVirtualItems() ?? [];
+    totalVerticalSize = $rowVirtualizer?.getTotalSize() ?? 0;
+    paddingTop = virtualRows?.length > 0 ? virtualRows?.[0]?.start || 0 : 0;
+    paddingBottom =
+      virtualRows?.length > 0
+        ? totalVerticalSize - (virtualRows?.at(-1)?.end || 0)
+        : 0;
+  }
+
+  $: columnVirtualizer = createVirtualizer({
+    horizontal: true,
+    count: colCt,
+    getScrollElement: () => container,
+    estimateSize: getColumnWidth,
+    overscan: 5,
+  });
+
+  let totalHorizontalSize = 0;
+  let paddingLeft = 0;
+  let paddingRight = 0;
+  let fixedVirtualColumns: VirtualItem[] = [];
+  let nonfixedVirtualColumns: VirtualItem[] = [];
+  $: {
+    const virtualColumns = $columnVirtualizer?.getVirtualItems() ?? [];
+    totalHorizontalSize = $columnVirtualizer?.getTotalSize() ?? 0;
+
+    // Manually calculate fixed virtual column set as they may not be in current virtualized item set
+    fixedVirtualColumns = range(fixedColCt).reduce((arr, idx) => {
+      const start = arr[idx - 1]?.end ?? 0;
+      const size = getColumnWidth(idx);
+      const end = start + getColumnWidth(idx);
+      return [
+        ...arr,
+        {
+          index: idx,
+          start,
+          end,
+          key: idx,
+          lane: 0,
+          size,
+        },
+      ];
+    }, []);
+
+    // If current virtual column set has fixed columns, remove them since we will use our measurements
+    nonfixedVirtualColumns = virtualColumns.filter(
+      (c) => c.index >= fixedColCt
+    );
+
+    paddingLeft =
+      virtualColumns?.length > 0 ? virtualColumns?.[0]?.start || 0 : 0;
+    // Adjust padding left for fixed virtual columns if needed
+    paddingLeft = Math.max(0, paddingLeft - fixedVirtualColumns.at(-1).end);
+    paddingRight =
+      virtualColumns?.length > 0
+        ? totalHorizontalSize - (virtualColumns?.at(-1)?.end || 0)
+        : 0;
+  }
+
+  const isFixedColumn = (idx: number) => idx < fixedColCt;
+</script>
+
+<div
+  bind:this={container}
+  style={`height: ${height}px; overflow: auto;`}
+  class="border"
+>
+  <table
+    class="border-collapse"
+    style={`height: ${totalVerticalSize}px; max-height: ${totalVerticalSize}px; width: ${totalHorizontalSize}px; max-width: ${totalHorizontalSize}px; overflow: none; table-layout: fixed`}
+  >
+    <thead class="sticky top-0 bg-gray-100 z-10">
+      <PivotVirtualRow element="th" {paddingLeft} {paddingRight}>
```

Comment:
Would it make sense to move the header row, padding row and the data rows in to their own component?

Reply:
No, I don't think adding a layer of abstraction gains us anything at this stage. This file is pretty comprehensible and maintainable as is IMO.
If complexity expands as we add features to the point that separating these segments into their own files would be beneficial, we can do that then.